.github/workflows/CI_build.yml

```
@@ -7,35 +7,113 @@ on:
 
 jobs:
 
-  analyze:
-    name: analyze / python-${{ matrix.python-version }} / ${{ matrix.os }}
+  first_check:
+    name: first code check / python-3.8 / ubuntu-latest
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
+      - name: Python info
+        run: |
+          which python
+          python --version
+      - name: Build package and create dev environment
+        run: |
+          python -m pip install --upgrade pip
+          pip install -e .[dev]
+      - name: Test with coverage
+        run: |
+          pytest --cov --cov-report term --cov-report xml --junitxml=xunit-result.xml
+      - name: Correct coverage paths
+        run: sed -i "s+$PWD/++g" coverage.xml
+      - name: Check style against standards using prospector
+        shell: bash -l {0}
+        run: prospector -o grouped -o pylint:pylint-report.txt
+      - name: Check whether import statements are used consistently
+        shell: bash -l {0}
+        run: isort --check-only --diff --conda-env matchms-dev --recursive --lines-after-imports 2 --force-single-line --no-lines-before FUTURE --no-lines-before STDLIB --no-lines-before THIRDPARTY --no-lines-before FIRSTPARTY --no-lines-before LOCALFOLDER matchms/ tests/ integration-tests/
+      - name: SonarCloud Scan
+        if: github.repository == 'matchms/matchms'
+        uses: sonarsource/sonarcloud-github-action@master
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+
+  build_pypi:
+    name: Pypi and documentation build / python-${{ matrix.python-version }} / ${{ matrix.os }}
```

Documentation is not tested here
```suggestion
    name: Pypi / python-${{ matrix.python-version }} / ${{ matrix.os }}
```